flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/ExecutingTest.java

```diff
@@ -252,28 +257,96 @@ public void testTransitionToFinishedOnSuspend() throws Exception {
     }
 
     @Test
-    public void testNotifyNewResourcesAvailableWithCanScaleUpTransitionsToRestarting()
+    public void testNotifyNewResourcesAvailableBeforeCooldownIsOverScheduledStateChange()
+            throws Exception {
+        try (MockExecutingContext ctx = new MockExecutingContext()) {
+            final Duration scalingIntervalMin =
+                    Duration.ofSeconds(1L); // do not wait too long in the test
+            final ExecutingStateBuilder executingStateBuilder =
+                    new ExecutingStateBuilder().setScalingIntervalMin(scalingIntervalMin);
+            Executing exec = executingStateBuilder.build(ctx);
+            exec.setLastRescale(Instant.now());
+            ctx.setCanScaleUp(true, null); // min met => rescale
+            ctx.setExpectRestarting( // scheduled rescale should restart the job after cooldown
+                    restartingArguments -> {
+                        assertThat(restartingArguments.getBackoffTime(), is(Duration.ZERO));
+                        assertThat(ctx.actionWasScheduled, is(true));
+                    });
+            exec.onNewResourcesAvailable();
+        }
+    }
+
+    @Test
+    public void testNotifyNewResourcesAvailableAfterCooldownIsOverStateChange() throws Exception {
+        try (MockExecutingContext ctx = new MockExecutingContext()) {
+            final ExecutingStateBuilder executingStateBuilder =
+                    new ExecutingStateBuilder().setScalingIntervalMin(Duration.ofSeconds(20L));
+            Executing exec = executingStateBuilder.build(ctx);
+            exec.setLastRescale(Instant.now().minus(Duration.ofSeconds(30L)));
+            ctx.setCanScaleUp(true, null); // min met => rescale
+            ctx.setExpectRestarting(
+                    restartingArguments -> { // immediate rescale
+                        assertThat(restartingArguments.getBackoffTime(), is(Duration.ZERO));
+                        assertThat(ctx.actionWasScheduled, is(false));
+                    });
+            exec.onNewResourcesAvailable();
+        }
+    }
+
+    @Test
+    public void testNotifyNewResourcesAvailableWithMinMetTransitionsToRestarting()
```

Comment:
And I see a lot of method name and comments include `MinMet`, `MinNotMet`, `min met` and `min not met`, etc.

Would you mind updating all of them to `withoutForce` or `withForce`?

Reply:
done. I agree that keeping high level semantics rather than implementation semantics is good but I feel like with this renaming we loose the real use case semantics that these tests are assessing: resources too small, resources lost etc...